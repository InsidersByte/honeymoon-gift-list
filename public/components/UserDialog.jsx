/* @flow */

import React from 'react';
import { Dialog, TextField } from 'material-ui';
import Form from './Form';
import ProgressButton from './ProgressButton';

type PropsType = {
  user: {
    email: string,
  },
  open: boolean,
  onHide: Function,
  onChange: Function,
  onSubmit: Function,
  saving: boolean,
};

const styles = {
  dialog: {
    maxWidth: 550,
  },
  input: {
    margin: '6px 0',
  },
  button: {
    marginTop: 12,
  },
};

export default function UserDialog({ user: { email }, open, onHide, onChange, onSubmit, saving }: PropsType) {
  return (
    <Dialog title="Invite a New User" open={open} onRequestClose={onHide} contentStyle={styles.dialog}>
      <Form onSubmit={onSubmit} loading={false} saving={saving}>
        <TextField name="email" type="email" floatingLabelText="Email" value={email} onChange={onChange} fullWidth style={styles.input} required />

        <ProgressButton saving={saving} label="Send Invitation Now" style={styles.button} fullWidth />
      </Form>
    </Dialog>
  );
}
