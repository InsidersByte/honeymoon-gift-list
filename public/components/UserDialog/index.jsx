/* @flow */

import React from 'react';
import { Dialog, TextField } from 'material-ui';
import Form from '../Form';
import ProgressButton from '../ProgressButton';
import type { AuthUser } from '../../types';

type PropsType = {
  user: AuthUser,
  open: boolean,
  saving: boolean,
  onHide: () => void,
  onChange: (event: SyntheticEvent) => void,
  onSubmit: (event: SyntheticEvent) => void,
};

const styles = {
  dialog: {
    maxWidth: 550,
  },
  input: {
    margin: '6px 0',
  },
  button: {
    marginTop: 12,
  },
};

const UserDialog = ({ user: { email }, open, onHide, onChange, onSubmit, saving }: PropsType) => (
  <Dialog title="Invite a New User" open={open} onRequestClose={onHide} contentStyle={styles.dialog}>
    <Form onSubmit={onSubmit} loading={false} saving={saving}>
      <TextField name="email" type="email" floatingLabelText="Email" value={email} onChange={onChange} fullWidth style={styles.input} required />

      <ProgressButton saving={saving} label="Send Invitation Now" style={styles.button} fullWidth />
    </Form>
  </Dialog>
);

export default UserDialog;
