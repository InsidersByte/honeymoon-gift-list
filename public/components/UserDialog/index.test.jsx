import React from 'react';
import { shallow } from 'enzyme';
import UserDialog from './';

const onHide = jest.fn();
const onChange = jest.fn();
const onSubmit = jest.fn();

const props = {
  user: { email: 'test@test.com' },
  open: true,
  saving: true,
  onHide,
  onChange,
  onSubmit,
};

describe('UserDialog', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<UserDialog {...props} />);

    expect(wrapper).toMatchSnapshot();
    expect(onHide).not.toHaveBeenCalled();
    expect(onChange).not.toHaveBeenCalled();
    expect(onSubmit).not.toHaveBeenCalled();
  });
});
