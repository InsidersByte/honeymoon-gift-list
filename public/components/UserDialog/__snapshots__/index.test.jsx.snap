// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDialog should render correctly 1`] = `
<Dialog
  autoDetectWindowHeight={true}
  autoScrollBodyContent={false}
  contentStyle={
    Object {
      "maxWidth": 550,
    }
  }
  modal={false}
  onRequestClose={[Function]}
  open={true}
  repositionOnUpdate={true}
  title="Invite a New User"
>
  <Form
    loading={false}
    onSubmit={[Function]}
    saving={true}
  >
    <TextField
      disabled={false}
      floatingLabelFixed={false}
      floatingLabelText="Email"
      fullWidth={true}
      multiLine={false}
      name="email"
      onChange={[Function]}
      required={true}
      rows={1}
      style={
        Object {
          "margin": "6px 0",
        }
      }
      type="email"
      underlineShow={true}
      value="test@test.com"
    />
    <ProgressButton
      fullWidth={true}
      label="Send Invitation Now"
      saving={true}
      style={
        Object {
          "marginTop": 12,
        }
      }
    />
  </Form>
</Dialog>
`;
